tap/tcp_reader.go

```diff
@@ -59,6 +60,7 @@ type tcpReader struct {
 	extension          *api.Extension
 	emitter            api.Emitter
 	counterPair        *api.CounterPair
+	sync.Mutex
```

Comment:
@nimrod-up9 there are two fields `isClosed` and `msgQueue` together needs to be guarded in concurrent calls to the following methods:

https://github.com/up9inc/mizu/blob/6a6575a72793f102311116edd237c797515420f2/tap/tcp_stream.go#L152-L170

https://github.com/up9inc/mizu/blob/6a6575a72793f102311116edd237c797515420f2/tap/tcp_stream.go#L184-L216

Since `tcpReader` struct is not exported there is no difference between embedding or not in terms of accessibility. There is also no performance difference as far as I know.